src/main/java/org/folio/circulation/domain/validation/ItemDeclaredLostValidator.java

```diff
@@ -0,0 +1,46 @@
+package org.folio.circulation.domain.validation;
+
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.results.CommonFailures.failedDueToServerError;
+
+import org.folio.circulation.domain.Item;
+import org.folio.circulation.domain.ItemStatus;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.ValidationErrorFailure;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+public class ItemDeclaredLostValidator {
```

Comment:
We now have three examples of refusing a request due to the item status, I think it is a good time to consolidate these implementations.

Either where we call the same method multiple times with each status, or a single invocation that takes a set of statuses.

Reply:
Yes, sure.
I did it. But, I consolidate only 2 classes (ItemDeclareLostValidator and ItemMissingValidator) in one (ItemStatusValidator) regarding item status check. I didn`t unite them with ItemNotFoundValidator and AlreadyCheckedOutValidator, because they have another, different mechanisms for the error messages, and this consolidation will involve changing in the error message response. Moreover, I don`t think that it is a great idea to check each status in  ItemStatusValidator for building separate error messages. 

What do you think about it?
Could you please review my approach?

Reply:
I can take a look, I imagined there were trade offs with each approach